File: C#/Projects/SnomedQuery.Parser/TRF2ItemGroup.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SnomedQuery.Parser
{
  /// <summary>
  /// Base class for RF2 group classes.
  /// </summary>
  /// <typeparam name="T"></typeparam>
  public class TRF2ItemGroup<T> : RF2Item
    where T : RF2ItemSingle, new()
  {
    /// <summary>
    /// Return curent active item. If current item is not active, return
    /// null.
    /// </summary>
    public T Active
    {
      get
      {
        T item = this.Current;
        if (item.Active == false)
          return null;
        return item;
      }
    }

    /// <summary>
    /// Return item that is currently active.
    /// Items can be added to snomed with an active date in the future.
    /// </summary>
    public T Current
    {
      get
      {
        Int32 index = this.items.Count - 1;
        while (index >= 0)
        {
          T item = this.items[index];
          if (item.EffectiveTime <= DateTime.Now)
            return item;
          index -= 1;
        }
        return null;
      }
    }

    /// <summary>
    /// List of items in group.
    /// </summary>
    public IEnumerable<T> Items
    {
      get { return this.items; }
    }

    /// <summary>
    /// List of items in group.
    /// </summary>
    protected List<T> items = new List<T>();
  }
}

File: Java/output/SnomedQuery/Parser/TRF2ItemGroup.java
package SnomedQuery.Parser;

import java.util.*;

  /** 
   Base class for RF2 group classes.
   
   <typeparam name="T"></typeparam>
  */
//C# TO JAVA CONVERTER TODO TASK: The C# 'new()' constraint has no equivalent in Java:
//ORIGINAL LINE: public class TRF2ItemGroup<T> : RF2Item where T : RF2ItemSingle, new()
  public class TRF2ItemGroup<T extends RF2ItemSingle> extends RF2Item
  {
	/** 
	 Return curent active item. If current item is not active, return
	 null.
	*/
	public final T getActive()
	{
	  T item = this.getCurrent();
	  if (item.Active == false)
	  {
		return null;
	  }
	  return item;
	}

	/** 
	 Return item that is currently active.
	 Items can be added to snomed with an active date in the future.
	*/
	public final T getCurrent()
	{
	  int index = this.items.size() - 1;
	  while (index >= 0)
	  {
		T item = this.items.get(index);
		if (item.EffectiveTime.compareTo(java.time.LocalDateTime.now()) <= 0)
		{
		  return item;
		}
		index -= 1;
	  }
	  return null;
	}

	/** 
	 List of items in group.
	*/
	public final Iterable<T> getItems()
	{
		return this.items;
	}

	/** 
	 List of items in group.
	*/
	protected ArrayList<T> items = new ArrayList<T>();
  }